tests/integration/base.py

```diff
@@ -278,6 +279,15 @@ def start_dnsmasq(self, ipv6_mode, iface):
         else:
             self.poll_text(dnsmasq_log, 'DHCP, IP range')
 
+    def iface_json(self, iface: str) -> dict:
+        '''Return iproute2's (detailed) JSON representation'''
+        out = subprocess.check_output(['ip', '-j', '-d', 'a', 'show', 'dev', iface],
+                                      text=True)
+        json_dict = json.loads(out)
+        if len(json_dict) == 1:
```

Comment:
suggestion: as empty arrays are evaluated to False, this could be simplified to `if json_dict:`